px-vis-markers.html

```diff
@@ -271,11 +271,12 @@
 
       for(var i=0; i<this.markerData.length; i++) {
         time = this.markerData[i]['time'];
+        datapoint = this.markerData[i]['datapoint'];
```

Comment:
You need to declare `datapoint` otherwise this becomes an implict global. Add `datapoint` to the variables declared between lines 266-270.

Reply:
updated the change in new commit.

Reply:
Sample MarkerData = [
  {
    "id": 1,
    "label": "Advisory",
    "time": 1505433900000,
    "datapoint": 1
  },
  {
    "id": 2,
    "label": "Advisory",
    "time": 1508025900000,
    "datapoint": 2
  },
  {
    "id": 3,
    "label": "Advisory",
    "time": 1510704300000,
    "datapoint": 3
  },
  {
    "id": 4,
    "label": "Advisory",
    "time": 1510963500000,
    "datapoint": 4
  },
  {
    "id": 5,
    "label": "Advisory",
    "time": 1511546100000,
    "datapoint": 5
  },
  {
    "id": 1,
    "label": "Adaptation-Successful",
    "time": 1505433900000,
    "datapoint": 1
  },
  {
    "id": 2,
    "label": "Outlier",
    "time": 1505434500000,
    "datapoint": 2
  },
  {
    "id": 2,
    "label": "VSG",
    "time": 1505434500000,
    "datapoint": 2
  },
  {
    "id": 3,
    "label": "Adaptation-Successful",
    "time": 1508026200000,
    "datapoint": 3
  }
]

Reply:
Sample Marker Config 
{
  "Advisory": {
    "location": "top",
    "color": "rgb(180,140,242)",
    "markerSymbol": "circle",
    "priority": 4,
    "markerSize": 25,
    "markerScale": 1,
    "markerFillOpacity": "1",
    "markerStrokeOpacity": 1,
    "row": 0,
    "timezone": "UTC",
    "firstDateTimeFormat": "hh:mm:ss A z"
  },
  "Adaptation-Successful": {
    "location": "bottom",
    "color": "rgb(222,82,136)",
    "markerSymbol": "circle",
    "priority": 5,
    "markerSize": 34,
    "markerScale": 1,
    "markerFillOpacity": "1",
    "markerStrokeOpacity": 1,
    "row": 2,
    "timezone": "UTC",
    "firstDateTimeFormat": "hh:mm:ss A z"
  }
}